### DISCUSSION TITLE
加三平方 = pow(? + 3, 2)

### DISCUSSION THREAD

**Discussion by nobodxbodon** [abersheeran/mingshe#19] [Ideas]
这种使用场景大概不多见？

BTW 木兰里可以这样：`加三平方 = n -> pow(n + 3, 2)`

**Comment by abersheeran** [comment]
这已经不是 partial 能做到的，需要用到 lambda。不过暂时对 lambda-shortcut 还没有特别好的思路，我觉得箭头函数并不适合 Python 里用，因为 typing 那边在讨论要出一个 `(int, int) -> int` 用来描述函数的语法，如果落实了，那么以后 `FunctionType = (int, int) -> int` 到底是一个匿名函数，还是一个函数类型定义就不好解析了。

**Comment by nobodxbodon** [comment]
hmm就这些个常用符号，用一个少一个呵。短期内倒是没有将木兰更新到3.7之后版本的打算。

之前就感到特殊符号在语言设计里的一些问题（就比如相对有限的易读的符号组合和越来越多的语言特性间的矛盾）。这两天正好 [在此答](https://www.zhihu.com/question/483359591/answer/2096999846) 写了点关于类自然语言语法的设想，有空的话请批评一下？

**Comment by abersheeran** [comment]
typing 在 3.7 之后就脱离了 Python 版本限制，在 3.7 里用 3.10 的 typing 完全是可以的。

**Comment by nobodxbodon** [comment]
确认一下，是指 [这个py3支持的指定函数返回类型语法](https://medium.com/@thomas_k_r/whats-this-weird-arrow-notation-in-python-53d9e293113) 吗？

刚看木兰可以用冒号：
```javascript
type Person {
    {
        name = "小白"
    }

    func $__init__(name) {
        self.name = name
    }
}

func shout(id): Person {
    return Person(id)
}

print(shout("Mulan1").name)
```

**Comment by abersheeran** [comment]
对。看起来你这个是木兰自己发明的语法？那倒还行。不过鸣蛇目的是做超集，所以不能不考虑 Python 原本的语法。

**Comment by nobodxbodon** [comment]
不知如何确保是超集呢？需要通过所有python测试用例集吗？

**Comment by abersheeran** [comment]
用的 Python3.9+ 同款解析器，同样的语法描述文件。
